### PR TITLE
Rewrite translator for Toronto Star

### PR DESCRIPTION
**Pull request by qubidt**
Not sure of the last time this translator was working, but it seems to be nonfunctional for the Toronto Star website as of time of writing. I rewrote the translator. Mostly uses the JSON-LD data because it seems to be the most consistent and accurate (e.g. doesn't truncate the description).

Some issues/assumptions:

* the hardcoded `.language = "en-CA"` was kept from the original translator, as were `.publicationTitle` and `.ISSN`. 
* `.libraryCatalog` was added to match the values from the original tests (not sure where this came from originally)
* based the translator off of the Embedded Metadata translator. This seems to be common but most of the data is hand-pulled so it might be more straightforward to skip this
* The tags don't match the original tests. Partially due to the embedded metadata translator and because the tags listed seem to have been removed from the pages

It passes the linter and the existing tests have been validated

### CODE REVIEW

File: Toronto Star.js [hunk 1]
```diff
@@ -56,10 +55,10 @@ function detectWeb(doc, url) {
 function getSearchResults(doc, checkOnly) {
 	var items = {};
 	var found = false;
-	var rows = ZU.xpath(doc, '//a[span[contains(@class, "story__headline")]]');
-	for (var i = 0; i < rows.length; i++) {
-		var href = rows[i].href;
-		var title = ZU.trimInternal(rows[i].textContent);
+	var rows = doc.querySelectorAll('a[data-test-id="mediacard"]');
```
Comment (AbeJellinek) [inline Toronto Star.js:58]: It looks like `'.c-search-results-item > a'` would be a better selector here. I'm always a little skeptical that these `data-test-id` attributes will stick around, since they're clearly meant to be internal.

File: Toronto Star.js [hunk 2]
```diff
@@ -56,10 +55,10 @@ function detectWeb(doc, url) {
...
+	for (let row of rows) {
+		let href = row.href;
+		let title = text(row, '*[data-test-id="mediacard-headline"]');
```
Comment (AbeJellinek) [inline Toronto Star.js:61]: This can just be `text(row, 'h3')`!

**Review by AbeJellinek — CHANGES_REQUESTED** [review]
Thanks, this is really, really great work!

Just a couple suggestions about selectors below. It would also be good to add a slash onto the end of the target, and you can feel free to add your name to the metadata/copyright if you'd like.

File: Toronto Star.js (hunk 1 above)
Comment (AbeJellinek) [reply Toronto Star.js:58]: Hmm, I see. I didn't realize the selector you had would work on other pages. That is worth keeping, then.

File: Toronto Star.js (hunk 2 above)
Comment (qubidt) [reply Toronto Star.js:61]: I was about to argue that it's unlikely that the headline is exclusively `h3` nor that `h3` are exclusively headlines. But from the   pages I tested this does seem to be the case, lol.

It also made me realize that the other selector missed some edge cases

File: Toronto Star.js (hunk 1 above)
Comment (qubidt) [reply Toronto Star.js]: I realized that the selector I had still missed some articles (see `.c-feature-mediacard` on [this page](https://www.thestar.com/topic.ottawa.html)). The new selector (`a[class*="-mediacard"]`) seems to be pretty consistent and also avoids the `data-test-id` attribute

**Comment by qubidt** [comment]
Issues have been fixed and added an edge case I missed

File: Toronto Star.js [hunk 3]
```diff
@@ -233,28 +266,76 @@ var testCases = [
...
 					{
-						"title": "Toronto Star Snapshot",
+						"title": "Snapshot",
 						"mimeType": "text/html"
 					}
 				],
-				"tags": [],
+				"tags": [
+					{
+						"tag": "CP_URGENCY_2"
```
Comment (AbeJellinek) [inline Toronto Star.js:306]: Sorry, last suggestion. Could we filter out tags that contain underscores? These clearly aren't meaningful in a Zotero library.

**Review by AbeJellinek — CHANGES_REQUESTED** [review]

File: Toronto Star.js (hunk 3 above)
Comment (qubidt) [reply Toronto Star.js]: Good idea! done

**Comment by AbeJellinek** [comment]
Perfect.
